src/pkg/services/m365/api/groups.go

```diff
@@ -135,10 +139,19 @@ func (c Groups) GetByID(
 		logger.CtxErr(ctx, err).Info("finding group by id, falling back to display name")
 	}
 
+	var filterTemplate string
+	// attempt to find by email address if the identifier looks like an email
+	if isEmail(identifier) {
```

Comment:
There should be an integration test in api/groups_test.go that needs updating to reflect these changes.

Reply:
done, thank you.